src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -687,6 +687,34 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/preview:
```

Comment:
Yeah that's a good point.  I will say, I don't necessarily mind preview and maybe we set a max value here (so ensure that we don't bring down the app.  For instance, nothing would be stopping someone from loading all HCA data into memory without the limit and that would make TDR very sad. I think a full access endpoint would look different (e.g. it might want to stream data out)

Reply:
>I think a full access endpoint would look different (e.g. it might want to stream data out)

I don't know how a streaming API would work; does openapi support streaming operations?

I was thinking of a paginated API, and in that case the it would look pretty much the same, except that it would accept `offset` as well as `count` (aka limit) and the result would include the total and remaining row count.

Reply:
> I was thinking of a paginated API, and in that case the it would look pretty much the same, except that it would accept offset as well as count (aka limit) and the result would include the total and remaining row count.

Are you thinking of `limit` and `offset`? That could be beneficial.

Reply:
I like the idea of turning it into a more general purpose API that can just return tables and paginate. To @nmalfroy 's concern, there already is a limit to the count so that it must be between 1 < n < 1000. If we turn it into a full pagination API, then I'd probably lower that limit to 100 rows, and API consumers would have to paginate.

Reply:
Cool.  I'm just thinking ahead to when we have to support this for parquet.

Reply:
Although do we have a suggestion of where those limits should be set? Right now they're just in BigQueryPdao. I guess the better long term solution would be to make it a configurable setting in our properties.

Reply:
I'd make an app config for the absolute max limit size but then make offset and limit params (the same as for enumerating datasets and snapshots and spend profiles

Reply:
(the query params, not the max page size

Reply:
I've added the pagination API :)